infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationServer.java

```diff
@@ -145,11 +161,23 @@ private void handleLrEntryRequest(@Nonnull RequestMsg request,
 
             LogReplicationSinkManager sinkManager = sessionToSinkManagerMap.get(session);
 
-            // If no sink Manager is found, drop the message and log an error
+            // We create a sinkManager for sessions that are discovered while bootstrapping LR. But as topology changes,
+            // we may discover new sessions. At the same time, its possible that the remote Source cluster finds a new
+            // session before the local cluster and sends a request to the local cluster.
+            // Since the two events are async, we wait to receive a new session in the incoming request.
+            // If the incoming session is not known to sessionManager drop the message (until session is discovered by
+            // local cluster), otherwise create a corresponding sinkManager.
+            // TODO[V2] : We still have a case where the cluster does not ever discover a session on its own.
+            //  To resolve this, we need to have a long living RPC from the connectionInitiator cluster which will query
```

Comment:
This is the case of client registration, right?  If the connection initiator is also the source of the replication data, registration will happen locally and there is no point for it to query the remote cluster, right?

Reply:
yup